src/tools/wasm2c-wrapper.hpp

```diff
@@ -0,0 +1,172 @@
+/*
+ * Copyright 2020 WebAssembly Community Group participants
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+//
+// Emit a C  wrapper file that can run the wasm after it is compiled with
+// wasm2c, useful for fuzzing.
+//
+
+namespace wasm {
+
+static std::string generateWasm2CWrapper(Module& wasm) {
+  std::string ret;
+  ret += R"(
+#include <stdint.h>
+#include <stdio.h>
+
+#include "wasm-rt-impl.h"
+#include "wasm.h"
+
+// Logging imports
+
+void _Z_fuzzingZ2DsupportZ_logZ2Di32Z_vi(u32 x) {
```

Comment:
It would be good to add a comment about why these names all have to be mangled and what they are for.